src/neural-network.js

```diff
@@ -307,42 +360,39 @@ export default class NeuralNetwork {
    * @param status { iterations: number, error: number}
    * @param options
    */
-  _checkTrainingTick(data, status, options) {
+  _trainingTick(data, status) {
     status.iterations++;
-    status.error = this._calculateTrainingError(data, options.learningRate);
+    status.error = this._calculateTrainingError(data);
 
-    if (options.log && (status.iterations % options.logPeriod === 0)) {
-      options.log(`iterations: ${status.iterations}, training error: ${status.error}`);
+    if (this.trainOpts.log && (status.iterations % this.trainOpts.logPeriod === 0)) {
+      this.trainOpts.log(`iterations: ${status.iterations}, training error: ${status.error}`);
     }
 
-    if (options.callback && (status.iterations % options.callbackPeriod === 0)) {
-      options.callback(Object.assign(status));
+    if (this.trainOpts.callback && (status.iterations % this.trainOpts.callbackPeriod === 0)) {
+      this.trainOpts.callback(Object.assign(status));
     }
   }
 
   /**
    *
    * @param data
-   * @param _options
+   * @param options
    * @returns {{error: number, iterations: number}}
    */
-  train(data, _options = {}) {
-    const options = Object.assign({}, this.constructor.trainDefaults, _options);
-    data = this.formatData(data);
-    options.log = this._setLogMethod(options.log);
-    options.learningRate = _options.learningRate || this.learningRate || options.learningRate;
-    var status = {
+  train(data, options = {}) {
+    this.updateTrainingOptions(options);
+    data = this._formatData(data);
+    const endTime = Date.now() + this.trainOpts.timeout;
```

Comment:
adding timeout?

Reply:
yeah, so they can train for a set time interval as if want.